### PR TITLE
Update and remover kudo 

### PR DESCRIPTION
**Pull request by mdobosz3**
An employee can edit a kudo they have given

An employee can see the form validation errors if they try to e.g., remove a given kudo's title

An employee can remove a kudo they have given

There's an edit and a destroy link on the kudos list next to each kudos given by the current employee

### CODE REVIEW

File: app/views/kudos/index.html.erb [hunk 1]
```diff
@@ -15,6 +16,13 @@
         <td><%= kudo.content %></td>
         <td><%= kudo.giver.email %></td>
         <td><%= kudo.receiver.email %></td>
+        <% if kudo.giver == current_employee %>
```
Comment (ciejj) [inline app/views/kudos/index.html.erb:19]: You are checking, if that kudo has been given by currently logged in user, but unfortunately  it it not enough.

Link to edit will look something like this:
`/kudos/1/edit
`
so I can put for example
`/kudos/2/edit`
in my browser, and access the edit page of a kudo, that does not belongs to me - try it out.

Same thing applies to delete - there should be additional check in the controller to make sure, that action can be performed.

File: app/controllers/kudos_controller.rb [hunk 2]
```diff
@@ -21,9 +29,26 @@ def create
...
+
   private
 
   def kudo_params
     params.require(:kudo).permit(:title, :content, :receiver_id)
   end
+
+  def kudo
+    @kudo ||= Kudo.find(params[:id])
+  end
```
Comment (ciejj) [inline app/controllers/kudos_controller.rb:53]: Memoization is great tool, but is it really useful here?
Each request generates new instance of the controller, so there is no way the `kudo` will be reused, and you will gain anything from applying this pattern

**Review by ciejj — CHANGES_REQUESTED** [review]
Looks good!
I suggest adding additional safeguard in cotrollers, and do some experiments with memoization

File: app/views/kudos/index.html.erb (hunk 1 above)
Comment (mdobosz3) [reply app/views/kudos/index.html.erb:19]: I increased the level of protection :)

File: app/controllers/kudos_controller.rb (hunk 2 above)
Comment (mdobosz3) [reply app/controllers/kudos_controller.rb:53]: You are right in this case, when a given value is not used often, it does not make sense.
I will try to create a TIL tomorrow

In addition, I noticed that I didn't have alerts made so I added them

**Review by ciejj — APPROVED** [review]
Good job!
+ congrats on finding, and fixing issues with notices on your own 👍

**Comment by oskarlakner** [comment]
🙌

**Pull request merged**